py_mmd_tools/check_mmd.py

```
@@ -0,0 +1,213 @@
+    Returns:
+        True / False
+    """
+
+    ok = True
+    for url in url_list:
+
+        try:
+            r = requests.get(url, timeout=10)
+            r.raise_for_status()
+            logger.debug(f'OK - {url}')
+            r.close()
+        except Exception as e:
+            logger.debug(f'NOK - {url}')
+            logger.debug(e)
+            ok = False
+
+    return ok
+
+
+def check_cf(cf_names):
+    """
+    Check that names are valid CF standard names
+    Args:
+        cf_names: list of names to test
+    Returns:
+        True / False
+    """
+    ok = True
+    for cf_name in cf_names:
+        try:
+            pti.get_cf_standard_name(cf_name)
+            logger.debug(f'OK - {cf_name} is a CF standard name.')
+        except IndexError:
+            logger.debug(f'NOK - {cf_name} is not a CF standard name.')
+            ok = False
+    return ok
+
+
+def check_vocabulary(xmldoc):
+    """
+    Check controlled vocabularies for elements:
+        - access_constraint
+        - activity_type
+        - operational_status (comment: also checked in MMD XSD schema)
+        - use_constraint
+    Args:
+        xmldoc: ElementTree containing the full XML document
+    Returns:
+        True / False
+
+    Comments: The following elements have test functions available in pythesint but are not used:
+    - area -> because it does not correspond to an element in currently tested files
+    - platform type -> because erroneous thesaurus in mmd repo?
+    """
+    vocabularies = {'access_constraint': 'access_constraints',
+                    'activity_type': 'activity_type',
+                    'operational_status': 'operstatus',
+                    'use_constraint': 'use_constraint_type',
+                    }
+    ok = True
+    for element_name, f_name in vocabularies.items():
+        elems_found = xmldoc.findall('./{*}' + element_name)
+        if len(elems_found) >= 1:
+            for rep in elems_found:
+                try:
```

same here - we could let it fail, couldn't we?